examples/data-objects/inventory-app/src/schema.ts

```diff
@@ -3,33 +3,23 @@
  * Licensed under the MIT License.
  */
 
-import {
-	AllowedUpdateType,
-	buildTreeConfiguration,
-	TypedNode,
-	SchemaBuilder,
-} from "@fluid-experimental/tree2";
+import { TreeConfiguration, SchemaFactory } from "@fluid-experimental/tree2";
 
-const builder = new SchemaBuilder({ scope: "com.contoso.app.inventory" });
+const builder = new SchemaFactory("com.contoso.app.inventory");
 
-export type Part = TypedNode<typeof Part>;
-export const Part = builder.object("Part", {
+export class Part extends builder.object("Part", {
 	name: builder.string,
 	quantity: builder.number,
-});
-
-export type Inventory = TypedNode<typeof Inventory>;
-export const Inventory = builder.object("Inventory", {
+}) {}
+export class Inventory extends builder.object("Inventory", {
 	parts: builder.list(Part),
-});
+}) {}
 
-export const treeConfiguration = buildTreeConfiguration({
-	schema: builder.intoSchema(Inventory),
-	allowedSchemaModifications: AllowedUpdateType.None,
-	initialTree: {
-		parts: {
-			// TODO: FieldNodes should not require wrapper object
-			"": [
+export const treeConfiguration = new TreeConfiguration(
+	Inventory,
+	() =>
```

Comment:
Note to self: does the initial tree state need to be deferred, or is this to avoid constructing the initial tree if we are not creating?

In either case, can the returned value be a Promise to facilitate dynamic imports, fetching a JSON template, etc.?

Reply:
You could also discourage re-use by making the second parameter a `NodeBuilderData`, which we pass to the schema constructor internally, rather than having the caller invoke the constructor. That would save the caller if they did something like:

```ts
const initialTree = new MyNode({...});
...
// TODO: I don't like reading documentation very much. Why is this a function? I'll worry about it later.
const config = new TreeConfiguration(MyNode, () => initialTree);
```
because instead they'd do:
```
const initialTree = {...};
...
// TODO: I don't like reading documentation very much. Why is this a function? I'll worry about it later.
const config = new TreeConfiguration(MyNode, () => initialTree);

It won't save them if there are more factory calls inside of the `new MyNode()`, but it's something. It might also discourage them more from doing the bad thing in the first place - because they might have trouble figuring out the type of `initialTree`. That said, this might just make the whole thing a little bit harder. Not sure - just an idea.

Reply:
I don't think that helps much. Nodes already implement NodeBuilderData in, so it would fail to give an error in the bad case, but it would also cause runtime errors if the type is ambiguous (for example if the root were a union, and you couldn't tell which type was desired from the data).

Also, putting object literals in lambdas is annoying since you need extra parentheses around them to parse correctly.

All that said, no reason not to allow all the insertable options for consistency with other code: I don't think it solves this node reuse issue, but its a general consistency improvement, so I have made the change.